### PR TITLE
 Filter packages that only contain ballerina source files when compiling

### PR DESCRIPTION
**Pull request by NatashaWso2**
## Automation tests
 - Integration tests
   > Added integration tests

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java [hunk 1]
```diff
@@ -144,12 +145,18 @@ public void execute() {
                 Path parentPath = sourcePath.getParent();
                 if (Files.isRegularFile(resolvedFullPath) && sourcePath.toString().endsWith(BLANG_SRC_FILE_SUFFIX) &&
                         parentPath != null) {
-                    Path fileName = parentPath.getFileName();
-                    String srcPkgName = fileName != null ? fileName.toString() : "";
                     throw LauncherUtils.createLauncherException("you are trying to build a ballerina file inside a " +
                                                                         "package within a project. Try running " +
                                                                         "'ballerina build <package-name>'");
                 }
+                // Checks if the source path is a package
```
Comment (sameerajayasoma) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java]: This check shouldn't be done here in this class. This is a low-level check that should be done somewhere else. 

We are calling the "ProjectDirs.containsSourceFiles" method here. It is a class that should only be known by the compiler not the build command.

File: cli/ballerina-launcher/src/main/java/org/ballerinalang/launcher/LauncherUtils.java [hunk 2]
```diff
@@ -115,6 +115,13 @@ public static void runProgram(Path sourceRootPath, Path sourcePath, String funct
                 throw createLauncherException("you are trying to run a ballerina file inside a package within a " +
                                                       "project. Try running 'ballerina run <package-name>'");
             }
+            // Checks if the source path is a package
```
Comment (sameerajayasoma) [inline cli/ballerina-launcher/src/main/java/org/ballerinalang/launcher/LauncherUtils.java]: This check shouldn't be done here in this class. This is a low-level check that should be done somewhere else.

We are calling the "ProjectDirs.containsSourceFiles" method here. It is a class that should only be known by the compiler not the build command.

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java (hunk 1 above)
Comment (NatashaWso2) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java]: Moved the check to the compiler with ada9c2d

File: cli/ballerina-launcher/src/main/java/org/ballerinalang/launcher/LauncherUtils.java (hunk 2 above)
Comment (NatashaWso2) [reply cli/ballerina-launcher/src/main/java/org/ballerinalang/launcher/LauncherUtils.java]: Moved the check to the compiler with ada9c2d

File: tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/PackageRunTestCase.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        String[] options = {"\n", "integrationtests\n", "\n", "f\n"};
+        balClient.runMain("init", new String[]{"-i"}, envVariables, options, new LogLeecher[0],
+                          projectPath.toString());
+    }
+
+    /**
+     * Create directory with a text file.
+     *
+     * @param path path of the directory to be created
+     * @throws IOException
```
Comment (anupama-pathirage) [inline tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/PackageRunTestCase.java]: Shall we add description here? Check all the places.

Comment (NatashaWso2) [reply tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/PackageRunTestCase.java]: Added descriptions in all the places with 5d30629

**Comment by NatashaWso2** [comment]
Closing this PR for now since the changes done should be reviewed again.

**Pull request closed without merging**